### PR TITLE
Update goto-checker documentation

### PR DESCRIPTION
**Pull request by peterschrammel**
Adds single-path symex checker and fault localization.

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-checker/README.md [hunk 1]
```diff
@@ -30,10 +30,15 @@ verification algorithm. Incremental goto checkers keep the state of the
...
-from the incremental goto checker (if it supports that).
-See \ref incremental_goto_checkert.
+verification algorithm has terminated.  See \ref incremental_goto_checkert.
+Incremental goto checkers can provide additional functionality by implementing
+the respective interfaces:
+* \ref goto_trace_providert: If a violated property has been
+  found then a trace can be retrieved from the incremental goto checker.
+* \ref fault_localization_providert: If a violated property has been
+  found then a likely fault location can be determined.
+* \ref witness_providert: A correctness witness can be output (for violated
```
Comment (tautschnig) [inline src/goto-checker/README.md]: None of these `\ref` appear to resolve according to CI.

Comment (tautschnig) [reply src/goto-checker/README.md]: Also: "correctness witness" and "violated property" don't go together.

File: src/goto-checker/README.md [hunk 2]
```diff
@@ -68,16 +73,36 @@ There are the following variants of goto verifiers at the moment:
   verification algorithm has terminated.
   The reporting uses 'goals' rather than 'property' terminology. I.e.
   a failing property means 'success' and a passing property 'failed'.
+* \ref stop_on_fail_verifier_with_fault_localizationt : Same as
+  \ref stop_on_fail_verifiert, but also finds an reports
```
Comment (tautschnig) [inline src/goto-checker/README.md]: s/an/and/

File: src/goto-checker/README.md [hunk 3]
```diff
@@ -68,16 +73,36 @@ There are the following variants of goto verifiers at the moment:
...
+  the most likely fault location. Requires an incremental goto checker
+  that provides fault localization.
+* \ref all_properties_verifier_with_fault_localizationt : Same as
+  \ref all_properties_verifier_with_trace_storaget, but also finds an reports
```
Comment (tautschnig) [inline src/goto-checker/README.md]: s/an/and/

**Review by tautschnig — APPROVED** [review]

File: src/goto-checker/README.md (hunk 1 above)
Comment (peterschrammel) [reply src/goto-checker/README.md]: Same trap again: there must be a space to separate the colon from the ref.

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: b901d2a77b6e89dde9b0b283f3f1af57b7d1d5c9).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/103662997
Status will be re-evaluated on next push.
Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

  - the compatibility was already broken by an earlier merge.

**Pull request merged**